lms/static/js/dashboard/views/course_view.js

```diff
@@ -0,0 +1,147 @@
+                    'click #block-course-msg a[rel="leanModal"]': 'unRegisterBlockCourse'
+                },
+
+                initialize: function (options) {
+                    this.setElement(this.el);
+                    this.listenTo(options.tabbedView, 'rendered', this.rendered);
+                },
+
+                rendered: function () {
+                    var $actionUnroll = this.$('.action-unenroll');
+                    var $unRegisterBlockCourse = this.$('#unregister_block_course');
+
+                    this.bindUnEnrollModal($actionUnroll);
+                    this.bindUnEnrollModal($unRegisterBlockCourse);
+                    this.bindEmailSettingsModal();
+                },
+
+                bindUnEnrollModal: function ($selector) {
+                    $selector.leanModal({
+                        overlay: 1,
+                        closeButton: ".close-modal"
+                    });
+
+                    var id = _.uniqueId('unenroll-');
+                    $selector.attr('id', id);
+                    var trigger = "#" + id;
+
+                    accessible_modal(
+                        trigger,
+                        "#unenroll-modal .close-modal",
+                        "#unenroll-modal",
+                        "#dashboard-main"
+                    );
+                },
+
+                bindEmailSettingsModal: function () {
+                    var $actionEmailSettings = this.$('.action-email-settings');
+
+                    $actionEmailSettings.leanModal({
+                        overlay: 1,
+                        closeButton: ".close-modal"
+                    });
+
+                    var id = _.uniqueId('email-settings-');
+                    $actionEmailSettings.attr('id', id);
+                    var trigger = "#" + id;
+
+                    accessible_modal(
+                        trigger,
+                        "#email-settings-modal .close-modal",
+                        "#email-settings-modal",
+                        "#dashboard-main"
+                    );
+                },
+
+                toggleCourseActionsDropdown: function (e) {
+                    var index = this.$(e.currentTarget).data('dashboard-index');
+
+                    // Toggle the visibility control for the selected element and set the focus
+                    var $dropDown = this.$('div#actions-dropdown-' + index);
+
+                    $dropDown.toggleClass('is-visible');
+                    if ($dropDown.hasClass('is-visible')) {
+                        $dropDown.attr('tabindex', -1);
+                    } else {
+                        $dropDown.removeAttr('tabindex');
+                    }
+
+                    // Inform the ARIA framework that the dropdown has been expanded
+                    var anchor = this.$(e.currentTarget);
+                    var ariaExpandedState = (anchor.attr('aria-expanded') === 'true');
```

Comment:
All vars should be declared at the top. Also, you are creating 2 jQuery objects with the current target in this function, you should only create one and utilize it multiple times.